packages/react/src/components/ChatHeader/ChatHeader.js

```diff
@@ -181,7 +194,7 @@ const ChatHeader = ({
       if (res.success) {
         setChannelInfo(res.room);
         if (res.room.t === 'p') setIsChannelPrivate(true);
-        if (res.room.ro) setMessageAllowed();
+        if (!res.room.ro) setMsgAndPinAllowed();
```

Comment:
Why have you negated the res.room.ro ? It was made for read only room. It might break the existing functionality of when to set the messages. Please make a different function to hide the pin message icon.

Reply:
I already fixed this in the last commit (fix readonly room [3935be0](https://github.com/RocketChat/EmbeddedChat/pull/520/commits/3935be06c498f5b14371e0136ff1c0bdd6b78764)), please check.
Also I already tried making a different function, but the API calls to get permissions were getting saturated very quickly as the API was called multiple times if a new function for pinning was made. So I decided that I can set canPinMsg from the same function itself.

Reply:
But now, how will disabling pinning messages work if the room is not set to read-only? Can you explain the flow ?

Reply:
> Because this function `if (res.room.ro) setMsgAndPinAllowed();` will only be called when the room is read-only. What about normal rooms? Earlier, while setting whether sending msg is allowed, it depended on whether the room is read-only or not, but in your case, it does not.

Before my commit, the code was `if (res.room.ro) setMessageAllowed();`. That means this also ran only when the room was set to read-only. For normal rooms, canSendMsg variable was already initialized as true, so didn't need to run the function. Now I have just added the canPinMsg variable in the same function and renamed the function to setMsgAndPinAllowed.



> If there is an issue making two functions, what you can do is this: always call `setMsgAndPinAllowed` without any condition and pass `res.room.ro` as an argument. Then, in the `setMsgAndPinAllowed` function, only run the `setMsg` part if the room is read-only. Also, make sure you refactor the code accordingly.

You are right, will do it right away!

Reply:
Reply for Edit: Instead of closing this PR, you can close the other one and mark this as ready for review once everything looks good. In the PR description, mention that this fixes both issues so that both will get merged right away.